src/Middleware/TrimMiddleware.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * Request middleware that trims all string values of PUT and POST requests.
+ * Some fields such as passwords are excluded.
+ */
+class TrimMiddleware implements MiddlewareInterface
+{
+    /**
+     * @var List of field names to exclude from trim
+     */
+    private const TRIM_EXCLUDE_LIST = ['password', 'password2'];
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        if (in_array($request->getMethod(), ['PUT', 'POST']) && is_array($request->getParsedBody())) {
+            $trimmedParsedBody = $this->trimArrayValues($request->getParsedBody());
+            $request = $request->withParsedBody($trimmedParsedBody);
+        }
+
+
+        return $handler->handle($request);
+    }
+
+    /**
+     * @template AK array key type
+     * @template AV array value type
+     * @param array<AK, AV> $array
+     * @return array<AK, AV>
```

Comment:
This is not a standard syntax and thus might break editors / parsers (as in: they won't show infos / autocompletion), i am not sure if using this is a good idea here?

Reply:
[It is kind of standard syntax](https://phpstan.org/blog/generics-in-php-using-phpdocs) and should be known by most editors.